File: Week-1/Data structures and Algorithms/Exercise-1/LoggerTest.java

public class LoggerTest {
    public static void main(String[] args) {
      
        Logger logA = Logger.getLogger();
        Logger logB = Logger.getLogger();

      
        if (logA == logB) {
            System.out.println("Logger instances are identical.");
        } else {
            System.out.println("Logger instances are different!");
        }

        System.out.println();

        runCredentialCheck(logA, logB);
    }
    
    private static void runCredentialCheck(Logger a, Logger b) {
        a.setCredentials("adminUser", "pass123");
        b.setCredentials("guestUser", "pass456");

        System.out.println("Logger A credentials:");
        a.printCredentials();

        System.out.println();

        System.out.println("Logger B credentials:");
        b.printCredentials();

        System.out.println("\nNote: Both loggers reflect the same state due to Singleton behavior.");
    }
}

File: Week-1/Design principles and  Patterns/Exercise-1/LoggerTest.cs
using System;

public class LoggerTest
{
    public static void Main(string[] args)
    {
        Logger logA = Logger.GetLogger();
        Logger logB = Logger.GetLogger();

        if (logA == logB)
        {
            Console.WriteLine("Logger instances are identical.");
        }
        else
        {
            Console.WriteLine("Logger instances are different!");
        }

        Console.WriteLine();

        RunCredentialCheck(logA, logB);
    }

    private static void RunCredentialCheck(Logger a, Logger b)
    {
        a.SetCredentials("adminUser", "pass123");
        b.SetCredentials("guestUser", "pass456");

        Console.WriteLine("Logger A credentials:");
        a.PrintCredentials();

        Console.WriteLine();

        Console.WriteLine("Logger B credentials:");
        b.PrintCredentials();

        Console.WriteLine("\nNote: Both loggers reflect the same state due to Singleton behavior.");
    }
}
